server/jetstream_cluster.go

```diff
@@ -7235,7 +7235,7 @@ func decodeStreamMsg(buf []byte) (subject, reply string, hdr, msg []byte, lseq u
 	if len(buf) < sl {
 		return _EMPTY_, _EMPTY_, nil, nil, 0, 0, errBadStreamMsg
 	}
-	subject = string(buf[:sl])
+	subject = bytesToString(buf[:sl])
```

Comment:
This will end-up being passed to processJetStreamMsg() which will call store's Store/StoreRawMsg with it and possibly store it in some map (like filestore's psim map). Also, even if you were to fix the usage of the subject/reply in the store implementation, this makes this a bit brittle since you have defined a Store interface that does not explicitly state the lifetime of those strings. In other words, the changes here (and below with `reply`) seem too risky. I would undo those.

Reply:
Decode owns the buf if you look at the call sites. I do understand your point though.